Utility/Utility.Public/Include/Utility/UserDataValidation.hpp

```diff
@@ -0,0 +1,53 @@
+#pragma once
+#include <regex>
+#include <string>
+
+namespace UserDataValidation
+{
+    /** @brief Check user's login.
+     * @params string login which contains user's login.
+     * @return true if login is valid.
+     * @details Login must have: /
+     *   -First char is a letter; /
+     *   -Lowercase letters; /
+     *   -Minimum 5 chars. /
+     * Example: login = 'some_nick34'.
+     */
+    inline bool isLoginValid(const std::string& login)
+    {
+        const std::regex pattern("([a-z]{1})([a-z0-9_-]{5,})");
+        return std::regex_match(login, pattern);
+    }
+
+    /** @brief Check user's email on valid.
+     * @params string email which contains user's email.
+     * @return true if email is valid.
+     * @details Email must: /
+     *   -have first char like letter; /
+     *   -maintain the following rule: /
+     *     Some letters + '@' + some letters + '.' + com/ru. /
+     * Example: email = 'Anton3_antonov4@epam.com'.
+     */
+    inline bool isEmailValid(const std::string& email)
+    {
+        const std::regex pattern(R"(([a-zA-Z]{1})(\w+@\w+\.)(?:com|ru))");
```

Comment:
Do you sure to use std::regax? It is maybe too slow.

Reply:
> Do you sure to use std::regax? It is maybe too slow.

Yes, checking for validaty take place on the client-side and the server is not loaded with it. But if checking was on the server-side, regex wouldn't be so good way, I think